pkg/mapper/iac-providers/arm/functions/concat.go

```diff
@@ -0,0 +1,58 @@
+/*
+    Copyright (C) 2020 Accurics, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package functions
+
+import (
+	"strings"
+
+	exp "github.com/VerbalExpressions/GoVerbalExpressions"
+	"go.uber.org/zap"
+)
+
+// Concat function splits str and runs respective functions on split parts.
+// Example: [Concat(parameters('vaultName'), '/', parameters('keyName'))]
+func Concat(vars, params map[string]interface{}, str string) string {
```

Comment:
consider this example: `[concat(parameters('namePrefix'), uniquestring(resourceGroup().id))]`

let say that the value of `namePrefix` is `x`, since we would not be able to resolve the uniqueString and resourceGroup() function, the result of above example would be `xuniquestring(resourceGroup().id)`.

I think it is not right, the result should have been `concat('x', uniquestring(resourceGroup().id))`.

let me know your thoughts on this.

Reply:
concat function can also be used for arrays? Would it work for arrays too? 
https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-array#concat

Reply:
Current scope is a bit limited for concat